experimental/dds/tree2/src/feature-libraries/contextuallyTyped.ts

```diff
@@ -234,6 +235,43 @@ export type ContextuallyTypedNodeData =
  */
 export type ContextuallyTypedFieldData = ContextuallyTypedNodeData | undefined;
 
+/**
+ * Information needed to interpret a subtree described {@link ContextuallyTypedNodeData} and {@link ContextuallyTypedFieldData}.
+ * @alpha
+ */
+export interface TreeDataContext {
+	/**
+	 * Schema for the document which the tree will be used in.
+	 */
+	readonly schema: SchemaDataAndPolicy;
+
+	/**
+	 * Procedural data generator for fields.
+	 * Fields which provide generators here can be omitted in the input contextually typed data.
+	 *
+	 * @remarks
+	 * TODO:
+	 * For implementers of this which are not pure (like identifier generation),
+	 * order of invocation should be made consistent and documented.
+	 * This will be important for identifier elision optimizations in tree encoding for session based identifier generation.
+	 */
+	getFieldGenerator(key: FieldKey, schema: FieldStoredSchema): undefined | FieldGenerator;
```

Comment:
I know I suggested naming this getFieldGenerator, and that name is pretty descriptive, but it makes things like defaultGetFieldGenerator and a GetFieldGenerator type kinda weird. I wonder if we can come up with a better name.

FieldGeneratorFactory? FieldProvider? Or maybe just FieldGenerators?

We could go another route and name it FieldSource.

Reply:
updated naming to `FieldSource`